go/src/meguca/server/init.go

```diff
@@ -245,3 +256,105 @@ func startServer() {
 		log.Fatal(err)
 	}
 }
+
+// checkGeoLite checks if the GeoLite2 country DB exists, and downloads if it doesn't
+func checkGeoLite() error {
+	_, err := os.Stat("GeoLite2-Country.mmdb")
+
+	if err != nil {
+		if os.IsNotExist(err) {
+			// Create the temporary archive and directory
+			tmpDir, err := ioutil.TempDir("", "tmp-")
+
+			if err != nil {
+				return err
+			}
+
+			defer os.RemoveAll(tmpDir)
+			tmp, err := ioutil.TempFile(tmpDir, "tmp-")
+
+			if err != nil {
+				return err
+			}
+			
+			// Get the archive itself and it's MD5 for security purposes
+			resp, err := http.Get("https://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz")
+
+			if err != nil {
+				return err
+			}
+
+			defer resp.Body.Close()
+			respMD5, err := http.Get("https://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz.md5")
```

Comment:
Do we even need this hash validation business?

Reply:
Yes, it's fairly light-weight and "ensures" you're not downloading something else.